pallets/settlement/src/lib.rs

```diff
@@ -1370,4 +1261,197 @@ impl<T: Trait> Module<T> {
+                        instruction_id,
+                        leg_id,
+                        LegStatus::ExecutionToBeSkipped(
+                            receipt.signer.clone(),
+                            receipt.receipt_uid,
+                        ),
+                    );
+                } else {
+                    if T::Portfolio::lock_tokens(
+                        &leg_details.from,
+                        &leg_details.asset,
+                        &leg_details.amount,
+                    )
+                    .is_err()
+                    {
+                        // rustc fails to infer return type of `with_transaction` if you use ?/map_err here
+                        return Err(DispatchError::from(Error::<T>::FailedToLockTokens));
+                    }
+                    <InstructionLegStatus<T>>::insert(
+                        instruction_id,
+                        leg_id,
+                        LegStatus::ExecutionPending,
+                    );
+                }
+            }
+            Ok(())
+        })?;
+
+        // Update storage
+        let affirms_pending = Self::instruction_affirms_pending(instruction_id)
+            .saturating_sub(u64::try_from(portfolios_set.len()).unwrap_or_default());
+        for portfolio in portfolios_set {
+            <UserAffirmations>::insert(portfolio, instruction_id, AffirmationStatus::Affirmed);
+            <AffirmsReceived>::insert(instruction_id, portfolio, AffirmationStatus::Affirmed);
+            Self::deposit_event(RawEvent::InstructionAffirmed(
+                did,
+                portfolio,
+                instruction_id,
+            ));
+        }
+
+        <InstructionAffirmsPending>::insert(instruction_id, affirms_pending);
+        for receipt in &receipt_details {
+            <ReceiptsUsed<T>>::insert(&receipt.signer, receipt.receipt_uid, true);
+            Self::deposit_event(RawEvent::ReceiptClaimed(
+                did,
+                instruction_id,
+                receipt.leg_id,
+                receipt.receipt_uid,
+                receipt.signer.clone(),
+            ));
+        }
+
+        // Execute instruction if conditions are met.
+        let execute_instruction_weight = Self::is_instruction_executed(
+            affirms_pending,
+            instruction_details.settlement_type,
+            instruction_id,
+        );
+
+        let execution_result = execute_instruction_weight.map(|info| {
+            info.actual_weight
+                .map(|w| {
+                    w.saturating_add(weight_for::weight_for_affirmation_with_receipts::<T>(
+                        u32::try_from(receipt_details.len()).unwrap_or_default(),
+                    ))
+                })
+                .into()
+        });
+
+        if ignore_execution {
+            return match execution_result {
+                Ok(post_info) => Ok(post_info),
+                Err(e) => Ok(e.post_info),
+            };
+        }
```

Comment:
Instead of modifying results here, let's send the correct result from this function. The function that is calling this function can decide if it wants to use the result or not.